### DISCUSSION TITLE
SSG ignore error from getStaticPaths in build if fetch fails (build without affected dynamic route)

### DISCUSSION THREAD

**Discussion by Tassil0** [vercel/next.js#42712] [Help]
I have a website that pulls data from two servers. The blog API is currently dead, therefore the build fails because Next fails to generate static paths for `/blog/[page]`. Is there a way to ignore or catch this error and build the website without the `/blog` dynamic route when the blog API is not available?
```
Error: Export encountered errors on following paths:
        /blog: /cs-cz/blog
        /blog: /en-gb/blog
```

Or should I approach this problem from a different perspective? The obvious solution would be to fix the blog API, but I don't have control over that. The blog API service has been unreliable in the past. The website should be able to build and update data without relying on the blog API.

**Comment by icyJoseph** [comment]
You could return and empty list of paths from `getStaticPaths`, and that'll post-pone all attempts at generating those pages to the runtime, where if they fail, nothing should change. Refer to https://nextjs.org/docs/basic-features/data-fetching/get-static-paths#generating-paths-on-demand
